crypto/rand/drbg_lib.c

```diff
@@ -394,8 +482,20 @@ int RAND_DRBG_generate(RAND_DRBG *drbg, unsigned char *out, size_t outlen,
         reseed_required = 1;
     }
 
-    if (drbg->reseed_counter >= drbg->reseed_interval)
-        reseed_required = 1;
+    if (drbg->reseed_interval > 0) {
+        if (drbg->generate_counter >= drbg->reseed_interval)
+            reseed_required = 1;
+    }
+    if (drbg->reseed_time_interval > 0) {
+        time_t now = time(NULL);
+        if (now < drbg->reseed_time
+            || now - drbg->reseed_time >= drbg->reseed_time_interval)
+            reseed_required = 1;
+    }
+    if (drbg->reseed_counter > 0 && drbg->parent != NULL) {
+        if (drbg->reseed_counter != drbg->parent->reseed_counter)
+            reseed_required = 1;
+    }
```

Comment:
I think it should also check that the parent's reseed time is up

Reply:
If we intend to support more than 2 levels, that would actually mean checking all parents.

Reply:
As it is currently, they have the same reseed interval by default (1 hour), so they'll both get reseed at that time. Do we care that the master doesn't get reseeded when it's time is up and only reseeds the next time that we actually use it? I guess not. So it's probably fine like it is.

Reply:
The design philosophy of the NIST DRBG standard is that every DRBG has its own reseeding strategy which does not depend on its parent's reseeding strategy. The parent DRBG is only used as entropy source, but there is no coupling between reseeding events.

The `reseed_counter` is a deviation from this principle. This is a compromise which was necessary to conciliate the different philosophies of the RAND_METHOD and the RAND_DRBG api. 

For the `reseed_interval` and the `reseed_time_interval` I don't intend to deviate from this simple principle. Since we have full control over reseed interval and reseed time interval of all three shared DRBGs, there is no need for the children to synchronize with the parent if we make reasonable choices for their default values.

The current defaults are just a first draft which maybe needs some finetuning.
```C
/* Default reseed intervals */
# define MASTER_RESEED_INTERVAL                  (1 << 8)
# define SLAVE_RESEED_INTERVAL                   (1 << 16)
# define MASTER_RESEED_TIME_INTERVAL             (60*60)   /* 1 hour */
# define SLAVE_RESEED_TIME_INTERVAL              (60*60)   /* 1 hour */
```
Maybe the slave reseed time interval should be much shorter and maybe it should not divide the master reseed interval, e.g.
```C
/* Default reseed intervals */
# define MASTER_RESEED_INTERVAL                  (1 << 8)
# define SLAVE_RESEED_INTERVAL                   (1 << 16)
# define MASTER_RESEED_TIME_INTERVAL             (60*60)   /* 1 hour */
# define SLAVE_RESEED_TIME_INTERVAL              (7*60)    /* 7 minutes */
```
In this matter I am open to and thankful for suggestions.

Reply:
The latter feels nicer but I've no solid evidence either way.  It reseeds the slaves more often which shouldn't hurt.

I don't see a benefit from the coprimality but it isn't going to hurt either.